app/src/main/java/de/tum/in/tumcampusapp/api/tumonline/converters/NullableEscapedStringConverter.kt

```diff
@@ -0,0 +1,36 @@
+package de.tum.`in`.tumcampusapp.api.tumonline.converters
+
+import android.os.Build
+import android.text.Html
+import com.tickaroo.tikxml.TypeConverter
+
+class NullableEscapedStringConverter : TypeConverter<String?> {
+
+    /**
+     * Parses leftover html escape sequences
+     *
+     * @param value String with leftover escape sequences
+     * @return unescaped String
+     */
+    private fun unescapeString(value: String): String {
+        // preserve newlines
+        val str = value.replace("\n", "<br />")
+
+        return if (Build.VERSION.SDK_INT >= 24) {
+            Html.fromHtml(str, Html.FROM_HTML_MODE_LEGACY).toString()
```

Comment:
Shouldn't this have an allow list of which HTML tags we actually want to support? Otherwise this might be a XSS entry point. Or am I missing something?

Reply:
As far as I am aware `Html.fromHtml` does not execute any javascript, does not support iframes and (since no handler is provided) does not load images. Any possible interactivity (like anker tags, which for example disappear completely) should be killed by the `toString()` anyways. Even if that fails and outbound connections were possible I see no possible way to make **authenticated** api calls from this environment, since there is no interaction with the java side of things.

This approach has the advantage that we don't have to rely on knowing all the escape sequences that might arise in the future. Also, this let's us rely on a proven parser implementation, instead of hacking our own, which would carry it's own xss risks.

Reply:
`Html.fromHtml` might not be executing the JS but the TextView that is displaying it might. In any case, I take your point and I think  you are right. This should't be a big deal.